src/content/blog/ga_qas.md

```diff
@@ -0,0 +1,233 @@
+    u = ansatz.g2(num_qubits = 3, num_layers = 1)
+    target_state = state.w(num_qubits = 3).inverse()
+).fit(num_steps = 100) # Define the optimization process and begin to optimize in 100 iterations
+compiler.plot() # Plot optimization process
+```
+
+**Make sure** that you can run the above code. We wrap it into a function $f: U \rightarrow \mathbb{R}$, the return value of this function is the (1 - cost value) of the optimization process, which means near 1 is good.
+
+```py
+import qiskit
+def fitnessW(qc: qiskit.QuantumCircuit):
+    qsp = QuantumStatePreparation(
+        u = qc,
+        target_state = state.w(num_qubits = 3).inverse()
+    ).fit()
+    return 1 - qsp.compiler.metrics['loss_fubini_study'][-1] # Fitness value
+```
+
+# Step 2. Configuration for genetic algorithm
+
+<figure>
+    <img src='/public/images/ga-qas/ga-qas2.png' style="margin: auto" alt='description' />
+    <figcaption>Fig 2. The general pipeline of GA-QAS</figcaption>
+</figure>
+
+A genetic algorithm (GA) is a heuristic algorithm based on a genetic combination process. GA processes include Fitness evaluation, Selection, Cross-over, and Mutation. We treat ansatz as an individual in the population (where gates are its genes).
+
+|        Process        |                                                                                                     Method                                                                                                      |
+| :-------------------: | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
+| Population generation |                                                A $n$ ansatz is initialized from a pool gate (Clifford$+R_i+CR_i, i\in\{x,y,z\}$) with certain provided metadata                                                 |
+|  Fitness evaluation   |                                                                             Evaluate each ansatz by calling `fitness_func(ansatz)`                                                                              |
+|       Selection       |                                                    Get fitness value from the fitness evaluation; simply sort from high to low and take the first half part                                                     |
+|      Cross-over       | Divide two ansatz $qc_1$ and $qc_2$ into four parts $\{qc_{11}, qc_{12}, qc_{21}, qc_{22}\}$, then combine each two parts into two new ansatz $qc_1^*=qc_{11}\vert qc_{22}$ and $qc_2^{*}=qc_{12}\vert qc_{21}$ |
+|       Mutation        |                                              Each gate on ansatz has a small probability of mutating to another gate (with the same number of qubits) in the pool                                               |
+
+_**Tab 1**. Detail of each operation in GA-QAS._
+
+The overall process can be viewed in _Fig. 2_.
```

Comment:
Line is out of context. Should it go somewhere else, or be removed?

Reply:
My mistake, the line `The overall process can be viewed in _Fig. 2_.` is moved to above paragraph.